contracts/strategies/JoeVoterProxy.sol

```diff
@@ -0,0 +1,326 @@
+     * @param _stakingContract Platypus Masterchef
+     * @param _pid PID
+     * @return pendingJoe
+     * @return bonusTokenAddress
+     * @return pendingBonusToken
+     */
+    function pendingRewards(address _stakingContract, uint256 _pid)
+        external
+        view
+        override
+        returns (
+            uint256 pendingJoe,
+            address bonusTokenAddress,
+            uint256 pendingBonusToken
+        )
+    {
+        (pendingJoe, bonusTokenAddress, , pendingBonusToken) = IJoeChef(_stakingContract).pendingTokens(
+            _pid,
+            address(voter)
+        );
+        uint256 reinvestFee = pendingJoe.mul(this.reinvestFeeBips()).div(BIPS_DIVISOR);
+
+        return (pendingJoe.sub(reinvestFee), bonusTokenAddress, pendingBonusToken);
+    }
+
+    function poolBalance(address _stakingContract, uint256 _pid) external view override returns (uint256 balance) {
+        return _poolBalance(_stakingContract, _pid);
+    }
+
+    /**
+     * @notice Pool balance
+     * @param _stakingContract Platypus Masterchef
+     * @param _pid PID
+     * @return balance in depositToken
+     */
+    function _poolBalance(address _stakingContract, uint256 _pid) internal view returns (uint256 balance) {
+        (balance, ) = IJoeChef(_stakingContract).userInfo(_pid, address(voter));
+    }
+
+    /**
+     * @notice Claim and distribute PTP rewards
+     * @dev Restricted to strategy with _pid
+     * @param _stakingContract Platypus Masterchef
+     * @param _pid PID
+     */
+    function claimReward(
+        uint256 _pid,
+        address _stakingContract,
+        address _extraToken
+    ) external override onlyStrategy(_stakingContract, _pid) {
+        voter.safeExecute(_stakingContract, 0, abi.encodeWithSignature("deposit(uint256,uint256)", _pid, 0));
+        _distributeReward(_extraToken);
+    }
+
+    /**
+     * @notice Distribute PTP and bonus rewards
+     * @dev Restricted to strategy with _pid
+     * @param _stakingContract Platypus Masterchef
+     * @param _pid PID
+     */
+    function distributeReward(
+        uint256 _pid,
+        address _stakingContract,
+        address _extraToken
+    ) external override onlyStrategy(_stakingContract, _pid) {
+        _distributeReward(_extraToken);
+    }
+
+    function _distributeReward(address _extraToken) private {
+        if (_extraToken == WAVAX) {
+            voter.wrapAvaxBalance();
+        }
+
+        uint256 pendingJoe = IERC20(JOE).balanceOf(address(voter));
+        uint256 pendingExtraToken = _extraToken > address(0) ? IERC20(_extraToken).balanceOf(address(voter)) : 0;
+        if (pendingJoe > 0) {
+            uint256 boostFee = 0;
+            if (boosterFee > 0 && boosterFeeReceiver > address(0) && voter.depositsEnabled()) {
+                boostFee = pendingJoe.mul(boosterFee).div(BIPS_DIVISOR);
+                voter.depositFromBalance(boostFee);
+                IERC20(address(voter)).safeTransfer(boosterFeeReceiver, boostFee);
```

Comment:
just to check, is the depositFromBalance always generating 1:1 tokens with what was deposited? I.e 100 joe generates 100 yyJoe?

overalll I think it's safer to use IERC20(address(voter)).balanceOf(this)

Reply:
That's correct, always mints 1:1.

I think it's safe as is, works without issue on Platypus. BalanceOf introduces additional gas costs.